openhands-sdk/openhands/sdk/llm/message.py

```diff
@@ -346,6 +346,9 @@ def _list_serializer(self) -> dict[str, Any]:
         if role_tool_with_prompt_caching:
             message_dict["cache_control"] = {"type": "ephemeral"}
 
+        if thinking_blocks_dicts:
+            message_dict["thinking_blocks"] = thinking_blocks_dicts
```

Comment:
@OpenHands please add a test to make sure when a thinking block exists, we put them as a field to `message_dict`